### PR TITLE
Add amazon-chime-sdk-component-library-react submodule

### PR DESCRIPTION
**Pull request by ltrung**
**Description of changes:**
Add `amazon-chime-sdk-component-library-react` submodule so that the [meeting demo](https://github.com/aws-samples/amazon-chime-sdk/tree/main/apps/meeting), which is often used to test development changes in `amazon-chime-sdk-component-library-react`, can use the latest changes from the library repo without waiting for npm release.

**Testing**

1. How did you test these changes? Manually run npm command.
2. Can these changes be tested using one of the demo application? If yes, which demo application can be used to test it? No

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apps/meeting/scripts/setup_chime_react_submodule.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    console.log(`Command ${command} failed with exit code ${cmd.status} signal ${cmd.signal}`);
+    console.log(cmd.stderr.toString());
+    process.exit(cmd.status);
+  }
+  return output;
+}
+
+process.chdir(path.join(__dirname, '../../../amazon-chime-sdk-component-library-react'));
+
+if (!fs.existsSync('node_modules')) {
```
Comment (devalevenkatesh) [inline apps/meeting/scripts/setup_chime_react_submodule.js:32]: Should we also handle, if the node_modules exists and there are changes in package.json then we need to re-install?

File: apps/meeting/scripts/setup_chime_react_submodule.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env node
```
Comment (devalevenkatesh) [inline apps/meeting/scripts/setup_chime_react_submodule.js:1]: In JS SDK [demo/scripts](https://github.com/aws/amazon-chime-sdk-js/tree/main/demos/browser/script) we follow a dashed naming convention, can we follow the same here: `setup-chime-react-submodule.js`?

File: apps/meeting/scripts/setup_chime_react_submodule.js (hunk 1 above)
Comment (devalevenkatesh) [reply apps/meeting/scripts/setup_chime_react_submodule.js:32]: May be do this everytime? Or have to add details to do so if any changes in react component library.

File: apps/meeting/package.json [hunk 3]
```diff
@@ -20,10 +20,13 @@
...
     "lint": "eslint src --ext .ts,.tsx --fix",
-    "build": "npm run lint && webpack --config ./webpack.config.js",
+    "preinstall": "scripts/setup_chime_react_submodule.js",
+    "build:fast": "webpack --config ./webpack.config.js",
+    "build": "npm run lint && npm install && npm run build:fast",
     "start:client": "webpack serve --config ./webpack.config.dev.js",
     "start:backend": "node server.js",
-    "start": "concurrently \"npm run start:client\" \"npm run start:backend\""
+    "start:fast": "concurrently \"npm run start:client\" \"npm run start:backend\"",
+    "start": "npm install && npm run start:fast"
```
[the remark this answers is not in the document]
Comment (simmkyu) [reply apps/meeting/package.json:29]: Yeah, I don't think running `npm install`, especially in the `start` script, is a good pattern.

File: apps/meeting/scripts/setup_chime_react_submodule.js (hunk 1 above)
Comment (ltrung) [reply apps/meeting/scripts/setup_chime_react_submodule.js:32]: Why do we need to reinstall? This is just for the submodule.

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[submodule "amazon-chime-sdk-component-library-react"]
+	path = amazon-chime-sdk-component-library-react
+	url = git@github.com:aws/amazon-chime-sdk-component-library-react.git
```
Comment (xuesichao) [inline .gitmodules:3]: The indention here seems to be not correct, could you please verify it?

Comment (ltrung) [reply .gitmodules:3]: This is auto-generated by git (i.e., I didn't touch it) :D

File: apps/meeting/scripts/setup_chime_react_submodule.js (hunk 1 above)
Comment (devalevenkatesh) [reply apps/meeting/scripts/setup_chime_react_submodule.js:32]: So you are changing into the `amazon-chime-component-library..` and then you are checking for `node_module` if not existing and then only run the rest commands, hence, what would be the case:
1. To update the component library to a newer commit?
2. If the newer commit contains a new dependency in component library and locally I have the `node_modules` already it wont update right?

Comment (ltrung) [reply apps/meeting/scripts/setup_chime_react_submodule.js:32]: You have to manually do it in that case anyway. This is purely a setup step to run it. I think it is the same way in JS SDK demo (we don't run npm install every time).

File: apps/meeting/scripts/setup_chime_react_submodule.js (hunk 2 above)
Comment (ltrung) [reply apps/meeting/scripts/setup_chime_react_submodule.js:1]: Fixed.

File: apps/meeting/scripts/setup_chime_react_submodule.js (hunk 1 above)
Comment (devalevenkatesh) [reply apps/meeting/scripts/setup_chime_react_submodule.js:32]: Ok, yeah I think my concern is just to let builders know how to update it :)

**Review by devalevenkatesh — APPROVED** [review]

**Review by xuesichao — APPROVED** [review]
Looks good to me.

**Review by xuesichao — APPROVED** [review]

**Pull request merged**